### PR TITLE
[SE-0069] Added UUID value type

### PR DESCRIPTION
**Pull request by colemancda**
Implements SE-0069 and potentially fixes https://bugs.swift.org/browse/SR-1756

Depends on https://github.com/apple/swift/pull/3095 to import UUID on Linux and other platforms.

### CODE REVIEW

File: Foundation/UUID.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
+//
+//  UUID.swift
+//  Foundation
+//
+//  Created by Alsey Coleman Miller on 6/20/16.
+//  Copyright © 2016 Apple. All rights reserved.
```
Comment (phausler) [inline Foundation/UUID.swift]: The copyright notice should be the same as the rest of the project

Comment (colemancda) [inline Foundation/UUID.swift]: Fixed

**Comment by colemancda** [comment]
@phausler With https://github.com/apple/swift/pull/3107 merged, this should compile fine for Linux. Can we get this tested and merged so I help out with other types affected by SE-0069?

**Comment by parkera** [comment]
@swift-ci please test

**Comment by phausler** [comment]
This also probably needs a reference in the build.py too for UUID.swift

**Comment by colemancda** [comment]
@phausler Done.

**Comment by parkera** [comment]
@swift-ci please test and merge

**Comment by phausler** [comment]
@swift-ci Please test

**Comment by phausler** [comment]
For some reason @parkera 's request was ignored by the build server. Not certain what happened.

**Comment by colemancda** [comment]
@phausler The build seems to be failing for a reason not related to this PR.

**Comment by phausler** [comment]
Yea I have been trying to isolate build failures all morning; and unfortunately I can't even get a working build myself right now. Once I can build this, I will merge it and also bring in the unit tests for the structural version.

**Comment by colemancda** [comment]
Can we try this again?

**Comment by phausler** [comment]
@swift-ci Please test

File: build.py [hunk 2]
```diff
@@ -375,6 +375,7 @@
 	'Foundation/NSXMLParser.swift',
 	'Foundation/FoundationErrors.swift',
 	'Foundation/URL.swift',
+    'Foundation/UUID.swift',
```
Comment (ianpartridge) [inline build.py]: Should be a tab.

**Comment by parkera** [comment]
@swift-ci Please test and merge

**Comment by parkera** [comment]
Looks like test & merge is still not enabled.

**Comment by parkera** [comment]
@swift-ci please test

**Comment by parkera** [comment]
Well, that failure seems wrong.

**Comment by colemancda** [comment]
Should I update from upstream?

**Comment by parkera** [comment]
Sure, let's give that a try and then I'll kick this off again.

Today is an awful day to be landing things; there is a lot of stuff going into Swift itself at the moment.

**Comment by colemancda** [comment]
Merged from swift-3.0-preview-3-branch, can we try again pls?

**Comment by parkera** [comment]
@swift-ci please test

**Comment by colemancda** [comment]
@parkera I dont think the fail is related to my code. Is there something I need to do from my side?

**Comment by parkera** [comment]
Sorry for the late reply; was out on vacation. Is it possible to rebase this PR on top of master? It now contains a lot of unrelated commits.

**Comment by mominul** [comment]
ping @colemancda

**Comment by parkera** [comment]
@swift-ci please test linux

**Comment by michael-lehew** [comment]
+1 to idea of rebase on top of swift-3.0-branch.  We can tease out the relevant parts, but it'd be nicer if the PR was isolated to just the changes relating to the UUID value type.

**Comment by colemancda** [comment]
Rebased in `swift-3.0-branch` in #554

**Pull request closed without merging**